tests/link/pytorch/test_basic.py

```diff
@@ -0,0 +1,187 @@
+    pytorch_mode=pytorch_mode,
+    py_mode=py_mode,
+):
+    """Function to compare python graph output and pytorch compiled output for testing equality
+
+    Parameters
+    ----------
+    fgraph: FunctionGraph
+        PyTensor function Graph object
+    test_inputs: iter
+        Numerical inputs for testing the function graph
+    assert_fn: func, opt
+        Assert function used to check for equality between python and pytorch. If not
+        provided uses np.testing.assert_allclose
+    must_be_device_array: Bool
+        Checks if torch.device.type is cuda
+
+
+    """
+    if assert_fn is None:
+        assert_fn = partial(np.testing.assert_allclose)
+
+    fn_inputs = [i for i in fgraph.inputs if not isinstance(i, SharedVariable)]
+
+    pytensor_torch_fn = function(fn_inputs, fgraph.outputs, mode=pytorch_mode)
+    pytorch_res = pytensor_torch_fn(*test_inputs)
+
+    if must_be_device_array:
+        if isinstance(pytorch_res, list):
+            assert all(isinstance(res, torch.Tensor) for res in pytorch_res)
+        else:
+            assert pytorch_res.device.type == "cuda"
+
+    pytensor_py_fn = function(fn_inputs, fgraph.outputs, mode=py_mode)
+    py_res = pytensor_py_fn(*test_inputs)
+
+    if len(fgraph.outputs) > 1:
+        for j, p in zip(pytorch_res, py_res):
+            assert_fn(j.cpu(), p)
+    else:
+        assert_fn([pytorch_res[0].cpu()], py_res)
+
+    return pytensor_torch_fn, pytorch_res
+
+
+@pytest.mark.parametrize("device", ["cpu", "cuda"])
+def test_pytorch_FunctionGraph_once(device):
+    """Make sure that an output is only computed once when it's referenced multiple times."""
+    from pytensor.link.pytorch.dispatch import pytorch_funcify
+
+    with torch.device(device):
+        x = vector("x")
+        y = vector("y")
+
+        class TestOp(Op):
+            def __init__(self):
+                self.called = 0
+
+            def make_node(self, *args):
+                return Apply(self, list(args), [x.type() for x in args])
+
+            def perform(self, inputs, outputs):
+                for i, inp in enumerate(inputs):
+                    outputs[i][0] = inp[0]
+
+        @pytorch_funcify.register(TestOp)
+        def pytorch_funcify_TestOp(op, **kwargs):
+            def func(*args, op=op):
+                op.called += 1
+                return list(args)
+
+            return func
+
+        op1 = TestOp()
+        op2 = TestOp()
+
+        q, r = op1(x, y)
+        outs = op2(q + r, q + r)
+
+        out_fg = FunctionGraph([x, y], outs, clone=False)
+        assert len(out_fg.outputs) == 2
+
+        out_torch = pytorch_funcify(out_fg)
+
+        x_val = torch.tensor([1, 2]).to(getattr(torch, config.floatX))
+        y_val = torch.tensor([2, 3]).to(getattr(torch, config.floatX))
+
+        res = out_torch(x_val, y_val)
+        assert len(res) == 2
+        assert op1.called == 1
+        assert op2.called == 1
+
+        res = out_torch(x_val, y_val)
```

Comment:
Same here

Reply:
The output (`res`) would be a pytorch tensor

Reply:
Not sure, perhaps these will which is fine, and SharedVariables wont. Anyway, just assert what they are